src/Tools/dotnet-counters/Exporters/ConsoleWriter.cs

```diff
@@ -184,6 +181,104 @@ public void AssignRowsAndInitializeDisplay()
+        {
+            List<(string header, string[] values)> observedTags = new();
+            List<int> columnHeaderLen = new();
+            List<int> maxValueColumnLen = new();
+            int tagsCount = 0;
+            foreach (ObservedTagSet tagSet in counter.TagSets.Values.OrderBy(t => t.Tags))
+            {
+                string[] tags = tagSet.DisplayTags.Split(',');
+                for (int i = 0; i < tags.Length; i++)
+                {
+                    string tag = tags[i];
+                    string[] keyValue = tag.Split("=");
+                    int posTag = observedTags.FindIndex (tag => tag.header == keyValue[0]);
+                    if (posTag == -1)
+                    {
+                        observedTags.Add((keyValue[0], new string[counter.TagSets.Count]));
+                        columnHeaderLen.Add(keyValue[0].Length);
+                        maxValueColumnLen.Add(default(int));
+                        posTag = observedTags.Count - 1;
+                    }
+                    observedTags[posTag].values[tagsCount] = keyValue[1];
+                    maxValueColumnLen[posTag] = Math.Max(keyValue[1].Length, maxValueColumnLen[posTag]);
+                }
+                tagsCount++;
+            }
+            AdjustColumnsLength(columnHeaderLen, maxValueColumnLen);
+            //print the header
+            string headerRow = "";
+            for (int i = 0; i < observedTags.Count; i++)
+            {
+                (string header, string[] values) observedTag = observedTags[i];
+                string headerWithSpaces = MakeFixedWidth(observedTag.header, Math.Max(maxValueColumnLen[i], columnHeaderLen[i]), truncateLeft: true);
+                headerWithSpaces += " ";
+                headerRow += headerWithSpaces;
+            }
+            if (!RenderCounterNameRow(ref row, headerRow, indentLevel: 2))
+            {
+                return false;
+            }
+            //print each line
+            int linePos = 0;
+            foreach (ObservedTagSet tagSet in counter.TagSets.Values.OrderBy(t => t.Tags))
+            {
+                tagSet.Row = row;
+                string tagRow = "";
+                for (int i = 0; i < observedTags.Count; i++)
+                {
+                    (string header, string[] values) observedTag = observedTags[i];
+                    string tagItem = observedTag.values[linePos];
+                    string tagWithSpaces = MakeFixedWidth(tagItem, Math.Max(maxValueColumnLen[i], columnHeaderLen[i]));
+                    tagWithSpaces += " ";
+                    tagRow += tagWithSpaces;
+                }
+                if (!RenderCounterValueRow(ref row, indentLevel: 2, tagRow, tagSet.LastValue, 0))
+                {
+                    return false;
+                }
+                linePos++;
+            }
+            return true;
+        }
+
+        private void AdjustColumnsLength(List<int> columnHeaderLen, List<int> maxValueColumnLen)
```

Comment:
It looks like we prioritize retaining the contents of values over the header names. Is that always the case for the information generated?

I'm wondering if there are scenarios in which it would make more sense to trim away parts of the longer values.
For instance, _if_ `http.route` weren't important (but in this case to me it looks the most important out of all the other fields), we could just trim away 15 char from the value and preserve the first 3 char of each other column
![dotnet_counters_example](https://github.com/dotnet/diagnostics/assets/16830051/51e3eae7-11f3-4a07-8801-ede5c41a178d)

I'm not famiilar with how many different stats would be logged, but are there any instances where the fields with above average character lengths are actually less important?

Reply:
In the extreme case, if `http.route` values were much longer, we would be seeing columns like 

```
ethod _code http.route                                          rsion cheme ntile
GET   200   api/............................................... 1.1   http  50
```

It makes me feel we should be able to trim outliers (extremely long fields), especially if the outliers are not that important.

Reply:
Thanks for the review @mdh1418 
I am not very familiar with dotnet-counters to be honest this was the first time that I saw it. As far as I understood we don't have control of what we will get as headers/values, so I can't decide to right trim the values because they can be useful and not repeated as in this sample.
Also the original issue the customer was complaining about truncating values and he couldn't see the end of the string which was important to him.
Also in this PR if the customer increase the width of his console nothing will be truncated, we removed the limit of 80 characters, if the customer didn't like the truncation he can expand the console and see the full content.
I don't have any strong opinion, if you/Noah decide that I should change the algorithm that I decided would be good based in the small experience that I have, I will be happy to change it.

Reply:
I also am not familiar with the typical user scenario for dotnet-counters, so maybe the trimmed header fields would be acceptable to preserve as much of the values as possible? I'll defer to participants of the issue @noahfalk @antonfirsov @JamesNK 

Another thing that popped to mind, I'm wondering how obvious is it to the user that they should increase the width of their console to see more of the headers if they can see the full values? Would we get issues in the future from folks reporting that header values are being truncated? This probably ties into whether we should prepend `...` or not.

Reply:
My suggestion is lets go ahead and ship what is here as-is and let users give us feedback on it. If we hear that the truncation needs to be made more obvious we can fine tune that in the future. In terms of optimizing exactly which information gets lost to truncation I suspect we've already reached the point where further complexity in the layout algorithm will have very diminished returns.